### ISSUE TITLE
Start the buildkit daemon on demand (socket-activated)

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#9947]
Buildkit uses a server (daemon), so trying to use the buildkit client fails if it is not started:

`error: failed to get status: rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /run/buildkit/buildkitd.sock: connect: no such file or directory"`

See #9640 (and https://github.com/moby/buildkit#quick-start)

Currently the `buildkitd` has to be started manually, as there is no systemd unit for it.

```console
$ sudo -b buildkitd
$ INFO[0000] auto snapshotter: using overlayfs            
WARN[0000] using host network as the default            
INFO[0000] found worker "p6bbje3hwqqbczwx909fjnaed", labels=map[org.mobyproject.buildkit.worker.executor:oci org.mobyproject.buildkit.worker.hostname:minikube org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/386] 
WARN[0000] using host network as the default            
INFO[0000] found worker "kgulhjepnr8a15fqgv0q4uag7", labels=map[org.mobyproject.buildkit.worker.executor:containerd org.mobyproject.buildkit.worker.hostname:minikube org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/386] 
INFO[0000] found 2 workers, default="p6bbje3hwqqbczwx909fjnaed" 
WARN[0000] currently, only the default worker can be used. 
INFO[0000] running server on /run/buildkit/buildkitd.sock
```

We should add a "buildkit.service" file for it, so that it can started/stopped/logged with systemd.

Similar to the current "containerd.service"

----

It would be nice if it was started automatically, using a systemd socket-activated service.

`sudo systemctl enable buildkit.socket`
`sudo systemctl start buildkit.socket`

This also allows setting group permissions, so that the socket can be accessed over a tunnel.

```
[Socket]
SocketMode=0660
SocketUser=root
SocketGroup=buildkit
```

Locally one can use `sudo`, but that doesn't work when using ssh tunnels to the unix socket.

`unix:///run/buildkit/buildkitd.sock`

`error: failed to get status: rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /run/buildkit/buildkitd.sock: connect: permission denied"`

Similar to the other runtimes:

* docker.sock #9229 #9232 
* podman.sock #9230 

Opened issues upstream:

* ~~https://github.com/moby/buildkit/issues/1923~~ 
* https://github.com/moby/buildkit/issues/1793

See http://0pointer.de/blog/projects/socket-activated-containers.html

**Comment by afbjorklund** [comment]
You can see how this will work, if you look at podman in the current release:

```                         _             _            
            _         _ ( )           ( )           
  ___ ___  (_)  ___  (_)| |/')  _   _ | |_      __  
/' _ ` _ `\| |/' _ `\| || , <  ( ) ( )| '_`\  /'__`\
| ( ) ( ) || || ( ) || || |\`\ | (_) || |_) )(  ___/
(_) (_) (_)(_)(_) (_)(_)(_) (_)`\___/'(_,__/'`\____)

$ ps aux | grep podman
docker     81090  0.0  0.0   3320   416 pts/1    S+   22:09   0:00 grep podman
$ podman --remote --url unix:///run/podman/podman.sock version
Client:
Version:      2.2.1
API Version:  2.1.0
Go Version:   go1.13.15
Git Commit:   a0d478edea7f775b7ce32f8eb1a01e75374486cb
Built:        Fri Dec 11 09:17:24 2020
OS/Arch:      linux/amd64

Server:
Version:      2.2.1
API Version:  2.0.0
Go Version:   go1.13.15
Git Commit:   a0d478edea7f775b7ce32f8eb1a01e75374486cb
Built:        Fri Dec 11 09:17:24 2020
OS/Arch:      linux/amd64
$ ps aux | grep podman
root       81104  1.0  0.5 884544 34816 ?        Ssl  22:09   0:00 /usr/bin/podman --log-level=info system service
docker     81178  0.0  0.0   3320   420 pts/1    S+   22:09   0:00 grep podman
```

So podman was **not** running, but when we accessed the socket it started up.

```
● podman.service - Podman API Service
     Loaded: loaded (/usr/lib/systemd/system/podman.service; static; vendor preset: enabled)
     Active: inactive (dead) since Sun 2020-12-13 22:09:34 UTC; 1min 26s ago
TriggeredBy: ● podman.socket
       Docs: man:podman-system-service(1)
```

```
● podman.socket - Podman API Socket
     Loaded: loaded (/usr/lib/systemd/system/podman.socket; enabled; vendor preset: enabled)
    Drop-In: /usr/lib/systemd/system/podman.socket.d
             └─override.conf
     Active: active (listening) since Sun 2020-12-13 14:43:26 UTC; 7h ago
   Triggers: ● podman.service
```

**Comment by afbjorklund** [comment]
Patch: https://patch-diff.githubusercontent.com/raw/moby/buildkit/pull/1924.patch

**Comment by afbjorklund** [comment]
Not available yet, in buildkit 0.8.x

**Comment by afbjorklund** [comment]
Seems to be working with 0.9.0, I opened a PR to set the mode to 660 instead of 666...

For minikube though, we probably want to add a group similar to the docker/podman ?
